services/provider/server/server.go

```diff
@@ -747,12 +729,20 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to update lastHeartbeat payload in the storageConsumer resource: %v", err)
 	}
 
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Failed to get the storageConsumer to set storage quota: %v", err)
+	}
+
+	clusterResourceQuotaSpec := s.getClusterResourceQuotaSpec(storageConsumer)
 	channelName, err := s.getOCSSubscriptionChannel(ctx)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
 	}
 
-	return &pb.ReportStatusResponse{DesiredClientOperatorChannel: channelName}, nil
+	desiredProviderConfigHash := util.GetMD5Hash(fmt.Sprintf("%v%s", clusterResourceQuotaSpec, channelName))
+
+	return &pb.ReportStatusResponse{DesiredProviderConfigHash: desiredProviderConfigHash}, nil
```

Comment:
The status-reporter is using the `channelName` to set an annotation on storageClient, if we pass a hash we might need to modify that

Reply:
yes i will take care in next updated commit.